ACC Algorithm/ACCAlgorithMain.c

```diff
@@ -0,0 +1,154 @@
+#include<stdio.h>
+
+//Globals
+float hostVel;
+float hostVelPrev; //Host velocity in the previous time step
+float integral = 0; //Needs to be global as it is used in the calculation of next step
+float errorPrev = -999.0; //Previous value of PID error
+float leadDistPrev = 0;
+float overshoot = 0;
+float leadTime = 0;
+
+//Calibratables
+float timeStep = 0.1; //Time step of system
+float simTime = 20.0; //Simulation time in seconds
+
+void vehDynamics(float torque){
+    //torque will be a value from -1 to 1, where negative values represent braking torque and positive values represent acceleration
+
+    //Calculate drag
+    //CdA = 0.27
+    float CdA = 0.55; //Default value for Drag Area
+    float rho = 1.293; //Default for air density
+    float mass = 1670.0; //Default of mass of vehicle
+
+    float drag = -0.5*rho*CdA*hostVel*hostVel/mass; //Deceleration due to drag
+
+    float hostAccel;
+    if (torque >= 0){ //Braking is more powerful than acceleration. Will this break the PID? Probably
+        hostAccel = drag + torque*4; //Assuming max acceleration is 4m/s; may want to make this a lookup table at some point
+    }
+    else if(torque < 0){
+        hostAccel = drag + torque*7; //Assuming max deceleration is 7m/s; may want to make this a lookup table at some point
+    }
+    
+    hostVel += hostAccel*timeStep; //Iterate the host velocity based on acceleration
+
+    //Prevent negative velocity
+    if (hostVel < 0){
+        hostVel = 0;
+    }
+}
+
+float accAlgo(float hostVelInit, float leadVel, float setSpeed, float leadDist){
+    //Decide between cruise and follow modes
+    int mode; //0 = cruise, 1 = follow
+    if (leadVel > setSpeed || leadDist > 500){ //Future additions: safety critical mode, failure mode
+        mode = 0;
+        //Do nothing, this algorithm defaults in cruise mode
+    }
+    else{
+        mode = 1;
+        setSpeed = leadVel;
+    }
+    
+    //Calculate PID terms
+    float error = setSpeed - hostVel; //P
+    integral += error*timeStep; //I
+    float derivative = (error - errorPrev)/timeStep; //D
+
+    //Set PID parameters
+    float kP = 3;
+    float kI = 0.05;
+    float kD = 0.04;
```

Comment:
can this be moved to its own function? would be nice to reuse pid functionality

ie.
```C
float pidStep(float error, pidParams* params) {
  ...
  return command;
}
```